src/business-logic/services/device-connection-service-implementation.ts

```diff
@@ -0,0 +1,106 @@
+import { DeviceConnectionService } from './interfaces/device-connection-service'
+import { Device } from '../../model/entities/device'
+import { DeviceConnectionFactory } from './interfaces/device-connection-factory'
+import { DeviceConnection } from './interfaces/device-connection'
+import { DeviceConnectionStatus } from '../../model/enums/device-connection-status'
+import { DeviceAlreadyConnectedException } from '../../model/exceptions/device-already-connected-exception'
+import { DeviceNotFoundException } from '../../model/exceptions/device-not-found-exception'
+import { Logger } from '@tv2media/logger/*'
+import { DeviceService } from './interfaces/device-service'
+
+type DeviceAggregate = {
+  deviceConnection: DeviceConnection,
+  device: Device
+}
+
+export class DeviceConnectionServiceImplementation implements DeviceConnectionService {
+  private readonly connectedDevices: Map<string, DeviceAggregate> = new Map()
+
+  constructor(private readonly deviceConnectionFactory: DeviceConnectionFactory, private readonly deviceService: DeviceService, private readonly logger: Logger) {
+    this.initialize(this.deviceService)
+      .then(() => {
+        logger.debug('Initialization of DeviceConnectionService done')
+      })
+      .catch((error) => {
+        logger.error(`Initialization of DeviceConnectionService failed with error: ${error}`)
+      })
+  }
+
+  private async initialize(deviceService: DeviceService): Promise<void> {
+    const devices: Device[] = await deviceService.getDevices()
+    await Promise.all(devices.map(device => this.createConnection(device)))
+  }
+
+  public async createConnection(device: Device): Promise<void> {
```

Comment:
This method is only called from within this class and from tests.
Since it's only called from within this class it should be private.
It looks like the only reason it's public is so we can call it from the tests which is an example of modifying production code solely becase of tests, which is a bad practice.

Reply:
Changed to private, removed from interface and instead i re-arranged the init-flow to avoid the hard coupling with the test-file.